http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpRequest.java

```diff
@@ -337,5 +363,17 @@ default Builder putHeader(String headerName, String headerValue) {
          * Removes all headers from this builder.
          */
         Builder clearHeaders();
+
+        /**
+         * Add a single un-encoded URI query parameter to be included in the created HTTP request.
+         *
+         * <p>This will <b>ADD</b> the value to any existing values already configured with this parameter name in
+         * the builder.</p>
+         *
+         * @param paramName  The name of the query parameter to add
+         * @param paramValue The un-encoded value for the query parameter.
+         */
+        Builder appendUriQueryParameter(String paramName, String paramValue);
```

Comment:
Hmm why do we need this instead of using `putRawQueryParameter`?

Reply:
since I wanted to make sure not to interfere with already in place param handling